### PR TITLE
Add settings to configure timeout options

### PR DESCRIPTION
**Pull request by barodeur**
Hello,

First of all thank you for this gem 🙏
We are using it at @TimelineInteractive to delay sending our event tracking using sidekiq.

We started to encounter some issues caused by timeouts of requests to segment. Since the ruby default timeout it 60 seconds, this caused serious delays in the processing of our queue.

We decided to fork the gem and configure the timeout options for the request.

### CODE REVIEW

File: lib/simple_segment/configuration.rb [hunk 1]
```diff
@@ -5,14 +5,16 @@ class Configuration
     include SimpleSegment::Utils
     include SimpleSegment::Logging
 
-    attr_reader :write_key, :on_error, :stub, :logger
+    attr_reader :write_key, :on_error, :stub, :logger, :open_timeout, :read_timeout
```
Comment (whatthewhat) [inline lib/simple_segment/configuration.rb]: To make this a little more generic I think we could introduce a `http_options` setting, that accepts a hash. We can pas it directly to Net::HTTP so people could use any of the `opt` values from https://docs.ruby-lang.org/en/2.0.0/Net/HTTP.html#method-c-start

```ruby
SimpleSegment::Client.new(
  write_key: 'key',
  http_options: {
    open_timeout: 42,
    read_timeout: 42,
    close_on_empty_response: true,
    # ...
  }
)
```

We'd just need to make sure that `use_ssl` is true by default.

Comment (barodeur) [reply lib/simple_segment/configuration.rb]: That's a great suggestion, I just implemented it in 7176580

File: spec/simple_segment/configuration_spec.rb [hunk 2]
```diff
@@ -22,6 +22,11 @@
       config = described_class.new(write_key: 'test')
       expect(config.on_error).to be_a(Proc)
     end
+
+    it 'has a default http_options' do
+      config = described_class.new(write_key: 'test')
+      expect(config.http_options).to eq({})
```
Comment (whatthewhat) [inline spec/simple_segment/configuration_spec.rb]: @barodeur Could we move the default options to the Configuration class, this spec would look like:

```ruby
config = described_class.new(write_key: 'test')
expect(config.http_options).to eq({ http_options: { use_ssl: true } })
```

Sorry for nitpicking! :)

Comment (barodeur) [reply spec/simple_segment/configuration_spec.rb]: no problem, I just moved the config to the configuration in 3ae2b4a

**Comment by whatthewhat** [comment]
@barodeur thanks for your contribution! I'll try to push this to rubygems next week.

**Pull request merged**